query-planner-js/src/buildQueryPlan.ts

```diff
@@ -742,7 +742,21 @@ function completeField(
     splitSubfields(context, fieldPath, subfields, subGroup);
     debug.groupEnd();
 
-    parentGroup.otherDependentGroups.push(...subGroup.dependentGroups);
+    // We need to hoist dependent groups of the subgroup to the parent group.
+    // In order to avoid duplicate fetches, we try to find existing dependent
+    // groups with the same service and merge path first.
+    for (const dependentGroup of subGroup.dependentGroups) {
+      const existingDependentGroup = parentGroup.otherDependentGroups.find(
+        (group) =>
+          group.serviceName === dependentGroup.serviceName &&
+          deepEqual(group.mergeAt, dependentGroup.mergeAt),
+      );
+      if (existingDependentGroup) {
+        existingDependentGroup.fields.push(...dependentGroup.fields);
```

Comment:
Shouldn't we merge other things than `fields` from `dependentGroup`? Mainly thinking of potential dependent groups of `dependentGroup`; I'm not sure we can guarantee those won't exists.

Reply:
Ah, good catch! I'll see if I can come up with a failing test, but I think you're right.